index.js

```diff
@@ -1,213 +1,130 @@
-const path = require('path');
-const findRoot = require('find-root');
-const glob = require('glob');
+const { dirname, resolve } = require('path');
+const { readFileSync } = require('fs');
+const pkgUp = require('pkg-up');
+const glob = require('glob-all');
+const webpackResolver = require('eslint-import-resolver-webpack');
+const webpack = require('webpack');
+const DirectoryNameAsMain = require('@elastic/webpack-directory-name-as-main');
 const debug = require('debug')('eslint-import-resolver-kibana');
 
 const defaults = {
   kibanaPath: '../kibana',
+  pluginDirs: [],
+  pluginPaths: [],
 };
 
 /*
  * Resolves the path to Kibana, either from default setting or config
  */
-function getKibanaPath(config, file, rootPath) {
+function getKibanaPath(config, rootPath) {
   const inConfig = config != null && config.kibanaPath;
 
   const kibanaPath = (inConfig)
-    ? path.resolve(config.kibanaPath)
-    : path.resolve(rootPath, defaults.kibanaPath);
+    ? resolve(config.kibanaPath)
+    : resolve(rootPath, defaults.kibanaPath);
 
   debug(`resolved kibana path: ${kibanaPath}`);
   return kibanaPath;
 }
 
-/*
- * Creates a glob pattern string that looks for:
- *  - file that matches the source (source.js)
- *  - directory with an index.js that matches the source (source/index.js)
- *  - directory with a matching module name that matches the source (source/source.js)
- * NOTE: last condition mentioned above can be removed when the custom resolver is removed from Kibana webpack config
- * @param {String|Array} source: the module identifier (./imported-file).
- */
-function getGlobPattern(source) {
-  if (Array.isArray(source)) {
-    const rootPath = path.join(...source);
-    const filename = source[source.length - 1];
-    return `./${rootPath}{*(.js),/${filename}.js,/index.js}`;
-  } else {
-    return `./${source}{*(.js),/${source}.js,/index.js}`;
-  }
-}
-
-/*
- * Returns an array of relative file path strings that match the source
- * @param {String} source: the module identifier
- * @param {String} checkPath: path to search in for file globbing
- */
-function getFileMatches(source, checkPath) {
-  const globPattern = getGlobPattern(source);
-  const globOptions = {
-    cwd: path.resolve(checkPath),
-  };
+function getProjectRoot(file, config) {
+  const { rootPackageName } = config;
 
-  const matches = glob.sync(globPattern, globOptions);
-  debug(`checking in ${checkPath}, matched ${matches.length}`);
+  const getRootPackageDir = (dir) => {
+    const pkgFile = pkgUp.sync(dir);
```

Comment:
Do we really need `pkgUp` for this? We're basically already doing what it is doing in this function.